scripts/translate-common.ts

```diff
@@ -17,6 +19,38 @@ export function traverse(path: string[], source: LocaleMap, result: ResultMap) {
         sv: _.get(sv, mpath),
         en: _.get(en, mpath),
       };
+    } else if (_.isNumber(source[key])) {
+      const mpath = _.join(path, '.') + '.' + key;
+      result[mpath] = {
+        path: mpath,
+        fi: source[key] as number,
+        sv: source[key] as number,
+        en: source[key] as number,
+      };
+    } else if (_.isArray(source[key])) {
+      (source[key] as Array<any>).forEach((item, index) => {
+        if (_.isString(item)) {
```

Comment:
There's a whole second level of decoding for array items. The traverse function can be altered to handle both arrays and objects to handle all cases in the same way:

```ts
export function traverse(path: string[], source: any, result: ResultMap) {
  if (_.isString(source)) {
    const mpath = _.join(path, '.');
    result[mpath] = {
      path: mpath,
      fi: source as string,
      sv: _.get(sv, mpath),
      en: _.get(en, mpath),
    };
  } else if (_.isNumber(source)) {
    const mpath = _.join(path, '.');
    result[mpath] = {
      path: mpath,
      fi: source as number,
      sv: source as number,
      en: source as number,
    };
  } else if (_.isArray(source)) {
    (source as Array<any>).forEach((item, index) => {
      traverse([...path, index.toString()], item, result);
    });
  } else if (_.isObject(source)) {
    for (const key in source) {
      traverse([...path, key], (source as LocaleMap)[key], result);
    }
  } else {
    throw new Error('Unknown type of ');
  }
}
```

Reply:
Thank you! I changed the code as you suggested.